main/main_timer_sync.cpp

```diff
@@ -147,10 +160,16 @@ MainFrameTime MainTimerSync::advance_core(float p_frame_slice, int p_iterations_
 
 // calls advance_core, keeps track of deficit it adds to animaption_step, make sure the deficit sum stays close to zero
 MainFrameTime MainTimerSync::advance_checked(float p_frame_slice, int p_iterations_per_second, float p_idle_step) {
+	if (p_idle_step <= 0) {
+		WARN_PRINT_ONCE("p_idle_step not positive");
+	}
+
 	if (fixed_fps != -1) {
 		p_idle_step = 1.0 / fixed_fps;
 	}
 
+	const float min_output_step = p_idle_step > 0 ? p_idle_step / 8 : 1E-6;
+
```

Comment:
```
min_output_step = p_idle_step / 8;
min_output_step = MAX(min_output_step, 1E-6);
```
Might be better, otherwise you could get small values of p_idle_step reaching zero.

Reply:
Right, that's safer. Unless we actually get a million FPS :)